aioitertools/builtins.py

```diff
@@ -159,7 +161,70 @@ def iter(itr: AnyIterable) -> AsyncIterator[T]:
     return value
 
 
-async def zip(*itrs: AnyIterable) -> AsyncIterator[Sequence[T]]:
+T1 = TypeVar("T1")
+T2 = TypeVar("T2")
+T3 = TypeVar("T3")
+T4 = TypeVar("T4")
+T5 = TypeVar("T5")
+
+# pylint: disable=unused-argument,missing-docstring,undefined-variable
+# pylint: disable=function-redefined,multiple-statements,too-many-arguments
+@overload
+def zip(__iter1: AnyIterable[T1]) -> AsyncIterator[Tuple[T1]]:
+    pass
+
+
+@overload
+def zip(
+    __iter1: AnyIterable[T1], __iter2: AnyIterable[T2]
+) -> AsyncIterator[Tuple[T1, T2]]:
+    pass
+
+
+@overload
+def zip(
+    __iter1: AnyIterable[T1], __iter2: AnyIterable[T2], __iter3: AnyIterable[T3]
+) -> AsyncIterator[Tuple[T1, T2, T3]]:
+    pass
+
+
+@overload
+def zip(
+    __iter1: AnyIterable[T1],
+    __iter2: AnyIterable[T2],
+    __iter3: AnyIterable[T3],
+    __iter4: AnyIterable[T4],
+) -> AsyncIterator[Tuple[T1, T2, T3, T4]]:
+    pass
+
+
+@overload
+def zip(
+    __iter1: AnyIterable[T1],
+    __iter2: AnyIterable[T2],
+    __iter3: AnyIterable[T3],
+    __iter4: AnyIterable[T4],
+    __iter5: AnyIterable[T5],
+) -> AsyncIterator[Tuple[T1, T2, T3, T4, T5]]:
+    pass
+
+
+@overload
+def zip(
+    __iter1: AnyIterable[Any],
+    __iter2: AnyIterable[Any],
+    __iter3: AnyIterable[Any],
+    __iter4: AnyIterable[Any],
+    __iter5: AnyIterable[Any],
+    __iter6: AnyIterable[Any],
+    *__iterables: AnyIterable[Any]
+) -> AsyncIterator[Tuple[Any, ...]]:
+    pass
```

Comment:
Is this overload meaningfully different from the core implementation below?  I feel like it could be dropped without losing any typing semantics.

Reply:
I added these overloads to make `zip()` conform to what is in typeshed for stdlib's `zip()`. And while it essentially the same as the implementation, once you add overloads, mypy ignores the implementation when calling the function.

Reply:
Good to know, I wasn't aware of that!